File: docs/examples/go/trigger_ping_check.go
// Command trigger_ping_check performs a manual ping run.
package main

import (
    "encoding/json"
    "fmt"
    "net/http"
)

const (
    triggerToken = "<token>"
    triggerProjectUUID = "<project-uuid>"
)

func main() {
    req, _ := http.NewRequest(http.MethodPost, baseURL+"/projects/"+triggerProjectUUID+"/ping-services/check", nil)
    req.Header.Set("Authorization", "Bearer "+triggerToken)

    resp, err := http.DefaultClient.Do(req)
    if err != nil {
        panic(err)
    }
    defer resp.Body.Close()

    if resp.StatusCode != http.StatusOK {
        panic(fmt.Sprintf("unexpected status: %d", resp.StatusCode))
    }

    var services []map[string]any
    if err := json.NewDecoder(resp.Body).Decode(&services); err != nil {
        panic(err)
    }

    fmt.Println("Ping results:", services)
}

File: docs/examples/python/trigger_ping_check.py
"""Trigger manual check for ping services."""

import requests

BASE_URL = "http://localhost:3000/api"
TOKEN = "<token>"
PROJECT_UUID = "<project-uuid>"


def trigger_ping_check() -> None:
    response = requests.post(
        f"{BASE_URL}/projects/{PROJECT_UUID}/ping-services/check",
        headers={"Authorization": f"Bearer {TOKEN}"},
        timeout=10,
    )
    response.raise_for_status()
    print("Updated ping services:", response.json())


if __name__ == "__main__":
    trigger_ping_check()
